run_unittests.py

```diff
@@ -2712,6 +2714,40 @@ def test_old_gnome_module_codepaths(self):
         self.build()
         mesonbuild.modules.gnome.native_glib_version = None
 
+    @unittest.skipIf(shutil.which('pkg-config') is None, 'Pkg-config not found.')
+    def test_pkgconfig_usage(self):
+        testdir1 = os.path.join(self.unit_test_dir, '24 pkgconfig usage/dependency')
+        testdir2 = os.path.join(self.unit_test_dir, '24 pkgconfig usage/dependee')
+        if subprocess.call(['pkg-config', '--cflags', 'glib-2.0'],
+                           stdout=subprocess.DEVNULL,
+                           stderr=subprocess.DEVNULL) != 0:
+            raise unittest.SkipTest('Glib 2.0 dependency not available.')
+        with tempfile.TemporaryDirectory() as tempdirname:
+            self.init(testdir1, ['--prefix=' + tempdirname, '--libdir=lib'], default_args=False)
+            self.install(use_destdir=False)
+            shutil.rmtree(self.builddir)
+            os.mkdir(self.builddir)
+            pkg_dir = os.path.join(tempdirname, 'lib/pkgconfig')
+            self.assertTrue(os.path.exists(os.path.join(pkg_dir, 'libpkgdep.pc')))
+            lib_dir = os.path.join(tempdirname, 'lib')
+            os.environ['PKG_CONFIG_PATH'] = pkg_dir
+            # Private internal libraries must not leak out.
+            pkg_out = subprocess.check_output(['pkg-config', '--static', '--libs', 'libpkgdep'])
+            self.assertFalse(b'libpkgdep-int' in pkg_out, 'Internal library leaked out.')
+            # Dependencies must not leak to cflags when building only a shared library.
+            pkg_out = subprocess.check_output(['pkg-config', '--cflags', 'libpkgdep'])
+            self.assertFalse(b'glib' in pkg_out, 'Internal dependency leaked to headers.')
+            # Test that the result is usable.
+            self.init(testdir2)
+            self.build()
+            myenv = os.environ.copy()
+            myenv['LD_LIBRARY_PATH'] = lib_dir
+            self.assertTrue(os.path.isdir(lib_dir))
+            #self.assertTrue(os.path.isfile(os.path.join(lib_dir, 'libpkgdep.so')))
```

Comment:
__[Flake8]__

[E265] block comment should start with '# '


<!-- Comment by [SideCI](https://sideci.com) -->